panacea/query_client.go

```diff
@@ -131,15 +166,26 @@ func (q QueryClient) GetStoreData(ctx context.Context, storeKey string, key []by
 		return nil, err
 	}
 
-	// get trustedBlock at blockHeight+1
-	time.Sleep(blockPeriod) // AppHash for query is in the next block, so have to wait until the next block is confirmed
-
-	// wait a creation of the next block
-	nextTrustedBlock, err := q.LightClient.VerifyLightBlockAtHeight(ctx, trustedBlock.Height+1, time.Now())
+	// get nextTrustedBlock from LightClient Primary.
+	// It returns a new light block on a successful update. Otherwise, it returns nil
+	nextTrustedBlock, err := q.LightClient.Update(ctx, time.Now())
 	if err != nil {
 		return nil, err
 	}
```

Comment:
If the next block does not exist, an error is occured and this process ends.
I think you need to make the code to retry a few times when an error that the block is not created occurs.

Reply:
As gyuguen said, I also think it's better to retry the `VerifyLightBlockAtHeight` a few times, instead of calling the `q.LightClient.Update(...)` two times in this function. We're calling the `q.LightClient.Update(...)` at the beginning of this function. So, the `q.LightClient.Update(...)` at line 171 will return `nil` with a very high probability. Then, we're gonna sleep 6 secs that decreases performance.

So, I would suggest something like:
```go
for {
  nextTrustedBlock, err = q.LightClient.VerifyLightBlockAtHeight(ctx, trustedBlock.Height+1, time.Now())
  if errors.Is(err, provider.ErrlightBlockNotFound) { // TODO: needs to be tested
    time.Sleep(1 * time.Second)
    continue
  } else if err != nil {
    return nil, err
  }
  break
}
```

Reply:
@youngjoon-lee @gyuguen Thanks, I fixed it in [07ede15](https://github.com/medibloc/panacea-doracle/pull/21/commits/07ede15f1487ce888f1e69000e6dc3c26eda21ec). Thank you for reviewing again.